WeatherForcast/WeatherForcast/Widgets/AnimationController.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  AnimationController.swift
+//  WeatherForcast
+//
+//  Created by Tung Tran on 10/7/19.
+//  Copyright © 2019 Sun. All rights reserved.
+//
+
+import UIKit
+
+class AnimationController: NSObject {
+    private let animationDuration: Double
+    private let animationType: AnimationType
+    
+    enum AnimationType {
+        case present
+        case dismiss
+    }
+    
+    init(animationDuration: Double, animationType: AnimationType ) {
+        self.animationDuration = animationDuration
+        self.animationType = animationType
+    }
+}
+
+extension AnimationController: UIViewControllerTransitioningDelegate {
+    
+}
+
+extension AnimationController: UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return TimeInterval(exactly: animationDuration) ?? 0
+    }
+    
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        guard let toVc = transitionContext.viewController(forKey: .to), let fromVc = transitionContext.viewController(forKey: .from) else {
+            transitionContext.completeTransition(false)
+            return
+        }
+        
+        switch animationType {
+        case .present:
+            transitionContext.containerView.addSubview(toVc.view)
+            presentAnimation(with: transitionContext, viewToAnimate: toVc.view)
+        case .dismiss:
+            transitionContext.containerView.addSubview(toVc.view)
+            transitionContext.containerView.addSubview(fromVc.view)
+            dismissAnimation(with: transitionContext, viewToAnimate: fromVc.view)
+        }
+        
+    }
+    
+    func dismissAnimation(with transitionContext: UIViewControllerContextTransitioning, viewToAnimate: UIView) {
+        let duration = transitionDuration(using: transitionContext)
+        let scaleDown = CGAffineTransform(scaleX: 0, y: 0)
+        let moveOut = CGAffineTransform(translationX: -viewToAnimate.frame.width, y: 0)
+        
+        UIView.animateKeyframes(withDuration: duration, delay: 0, options: .calculationModeLinear, animations: {
+            
+            UIView.addKeyframe(withRelativeStartTime: 0, relativeDuration: 0.4, animations: {
```

Comment:
```suggestion
            UIView.addKeyframe(withRelativeStartTime: 0, relativeDuration: 0.4 ) {
```
dùng strailing closure